src/index.d.ts

```diff
@@ -114,15 +115,23 @@ export type TVCompoundSlots<
       } & ClassProp
 >;
 
-export type TVDefaultVariants<V extends TVVariants<S>, EV extends TVVariants, S extends TVSlots> = {
-  [K in keyof V | keyof EV]?: StringToBoolean<keyof V[K] | keyof EV[K]>;
+export type TVDefaultVariants<
+  V extends TVVariants<S>,
+  S extends TVSlots,
+  EV extends TVVariants<ES>,
+  ES extends TVSlots,
+> = {
+  [K in keyof V | keyof EV]?:
+    | (K extends keyof V ? StringToBoolean<keyof V[K]> : never)
+    | (K extends keyof EV ? StringToBoolean<keyof EV[K]> : never);
 };
 
 export type TVScreenPropsValue<
-  V extends TVVariants,
+  V extends TVVariants<S>,
+  S extends TVSlots,
   K extends keyof V,
   C extends TVConfig,
-> = isStringArray<C["responsiveVariants"]> extends true
```

Comment:
`isStringArray` was not properly narrowing the type, so inlining it here fixes the error.